tests/4.5/taskloop/test_taskloop_num_tasks.c

```diff
@@ -0,0 +1,76 @@
+//------------------------------------------- test_taskloop_num_tasks.c --------------------------------------//
+//
+// OpenMP API Version 4.5 September 2015
+//
+// This test checks the 'taskloop' directive with the 'num_tasks' clause specified.
+// The 'taskloop' construct parallelize loops with independent iterations by creating tasks. 
+// It allows for efficient parallel execution of loop iterations by distributing them among multiple threads. 
+// The 'num_tasks' clause variable ensures that the loop iterations are shared amoung created 'num_tasks'. 
+// i.e the loop iterations are divided amoung number of tasks (each num_task group is run by same the thread).   
+//----------------------------------------------------------------------------------------------------------//
+
+#include <stdio.h>
+#include <omp.h>
+#include "ompvv.h"
+
+#define NUM_THREADS 100
+#define NUM_TASKS 4
+
+int test_taskloop_num_tasks() {
+
+  int errors = 0;
+
+  long int all_thread_sum, real_sum = 0; 
+  long int var = 0; //This variable is shared with all the tasks.  
+  
+  int NUM = 100;
+
+   //get valid sum without openmp
+   for(int i = 0; i < NUM; i++) {
+     real_sum = real_sum + i;
+   }
+
+   int thread_ids[100];
+   int values[100];
+
+   #pragma omp parallel num_threads(NUM_THREADS)
```

Comment:
A thought, if NUM_THREADS = NUM_TASKS and if we can check if at least 1 task was executed by a different thread (need not be 25 each) then we can kind-of conclude that the num tasks worked.

Reply:
Yes, the 'NUM_TASKS' loop iterations are run by different threads. But, to compare the value of expression with openmp and without openmp, 'NUM_TASKS' times values are used with 'NUM_THREADS/NUM_TASKS'.

Reply:
Can you explain how the test checks NUM_TASKS ?

Reply:
![image](https://github.com/SOLLVE/sollve_vv/assets/139203039/44ecedd6-0916-4ffe-9814-d0cd0c2b0026)

@line 17, NUM_TASKS=4 and @line 26, NUM=12.
@line 40, the NUM_TASKS variable chunks the 12 loop iterations into 4 groups of tasks. 
Tasks Group 1: [0, 1, 2] by thread_id 74 
Tasks Group 2: [3, 4, 5] by thread_id 65 
Tasks Group 3: [6, 7, 8] by thread_id 73
Tasks Group 4: [9, 10, 11] by thread_id 0
The tasks present in each group is run by same thread.
With openmp taskloop ‘num_tasks’ clause, the sum of all the values is 66.  This value is tested against the ‘sum’ value calculate without openmp for NUM.  To avoid race condition, used ‘atomic’ for ‘var’ variable.
 
![image](https://github.com/SOLLVE/sollve_vv/assets/139203039/de806a4c-aa29-49d1-bd9e-2393d7436fd8)

Reply:
Yes, the sum would not change if the loop was executed by 1 or more threads OR the number of generated tasks was not as expected.

Reply:
Your expecting to check that is the (num/num_tasks) tasks were each executed by the same thread or not? and remove the ‘atomic’ for ‘var’ variable. Is my understanding correct?

Reply:
The atomic can stay; the test needs to record the number of iterations executed by each thread. Following is correct behavior: 1)If the same thread executes it and the number of threads in the region is 1 and 2) If there are multiple threads and each execute a (num/num_tasks) tasks.

Reply:
@spophale, please have a look at https://github.com/SOLLVE/sollve_vv/commit/555ff11959959ddcbe4379ddb84e8ceff9f90b52